pandas/core/frame.py

```diff
@@ -7614,8 +7627,10 @@ def _get_data(axis_matters):
                     # TODO: combine with hasattr(result, 'dtype') further down
                     # hard since we don't have `values` down there.
                     result = np.bool_(result)
-            except TypeError:
+            except (TypeError, ValueError):
                 # e.g. in nanops trying to convert strs to float
+                # TODO: the ValueError is raised in trying to convert str
```

Comment:
yes this *should* be a TypeError, where is the ValueError coming from?

Reply:
what is trying to convert to str?

Reply:
do you still need this ValueError? (as i think you changed below)

Reply:
i'll re-run without catching ValueError.  IIRC it was in nanops._ensure_numeric

Reply:
yah, there are 3 tests where `x.astype(np.float64)` raises a ValueError in _ensure_numeric with a object-dtype `x` that contains a string.

Reply:
just removed this and re-raised as TypeError on nanops 1292-1296